src/rdkafka_metadata.c

```diff
@@ -1971,3 +1971,179 @@ rd_kafka_metadata_new_topic_with_partition_replicas_mock(int replication_factor,
+                                } else {
+                                        rd_kafka_log(
+                                            rk, LOG_WARNING, "METADATAUPDATE",
+                                            "Partition %s [%d]: new leader "
+                                            "%" PRId32 " not found in cache",
+                                            topic, part, mdp->leader);
+                                }
+                                continue;
+                        }
+
+                        current_leader_epoch =
+                            rkmce->rkmce_metadata_internal_topic
+                                .partitions[part]
+                                .leader_epoch;
+
+                        if (current_leader_epoch >= mdpi->leader_epoch) {
+                                rd_kafka_broker_destroy(rkb);
+                                if (by_id) {
+                                        rd_kafka_dbg(
+                                            rk, METADATA, "METADATAUPDATE",
+                                            "Partition %s [%d]: leader epoch "
+                                            "is "
+                                            "not newer %" PRId32 " >= %" PRId32,
+                                            rd_kafka_Uuid_base64str(&topic_id),
+                                            part, current_leader_epoch,
+                                            mdpi->leader_epoch);
+                                } else {
+                                        rd_kafka_dbg(
+                                            rk, METADATA, "METADATAUPDATE",
+                                            "Partition %s [%d]: leader epoch "
+                                            "is "
+                                            "not newer %" PRId32 " >= %" PRId32,
+                                            topic, part, current_leader_epoch,
+                                            mdpi->leader_epoch);
+                                }
+                                continue;
+                        }
+                        cache_updated = rd_true;
+                        partition_cache_changes++;
+
+                        /* Need to acquire the write lock to avoid dirty reads
+                         * from other threads acquiring read locks. */
+                        rd_kafka_wrlock(rk);
+                        rkmce->rkmce_metadata_internal_topic.partitions[mdp->id]
+                            .leader_epoch = mdpi->leader_epoch;
+                        rkmce->rkmce_mtopic.partitions[mdp->id].leader =
+                            mdp->leader;
+                        rd_kafka_wrunlock(rk);
+                        rd_kafka_broker_destroy(rkb);
+
+                        if (by_id) {
+                                rd_kafka_dbg(rk, METADATA, "METADATAUPDATE",
+                                             "Partition %s [%d] "
```

Comment:
Change this placeholder to match the type, here and in other places in this function
```suggestion
                                             "Partition %s [%" PRId32 "] "
```